app/src/main/java/edu/outside2154/gamesense/model/Stat.kt

```diff
@@ -10,26 +15,42 @@ class Stat (initGoals: Map<String, Double>) {
         }
     }
 
-    var goals = StatItems(initGoals.mapValues { (_, v) -> v * 60.0 })
-        private set
-    var current = StatItems(initGoals.mapValues { (_, v) -> v * 0.0 })
-        private set
+    var goals = StatItems(initGoals)
+    var current = StatItems(currGoals)
 
     fun updateCurrent(data: Map<String, Double>) {
-        current += StatItems(data.mapValues { (_, v) -> v * 60.0 })
+        current += StatItems(data)
     }
 
     fun calcStat(): Double? {
         val divisor = goals.items.values.sum()
         if (divisor == 0.0) return null
 
         val elements = current.items.mapValues { (k, v) ->
-            minOf(v, goals.items[k] ?: 0.0)
+            minOf(v, goals.items[k] ?: Double.MAX_VALUE)
```

Comment:
This is possibly more semantically correct:

```kotlin
val elements = goals.items.mapValues { current.items[it.key] ?: 0.0 }
```

Reply:
This one we are trying to avoid overflow so we have to use minOf